tests/Test/IntegrationCase.php

```diff
@@ -105,7 +105,12 @@ public function getInputCode(): ?string
         return $this->inputCode;
     }
 
-    public function getRequirement(string $name): int
+    /**
+     * @return array<string>|int
+     *
+     * @phpstan-return ($name is 'php' ? int : array<string>)
+     */
+    public function getRequirement(string $name)
```

Comment:
plus property `requirements`, plus `getRequirements` method

also, we have `php<` key

probably I didn't cover everything

Reply:
Can I just change return type to `mixed` 😉? This generic method is just wrong, it should be `getPhpRequirement()`, but I didn't want to change it within this PR.

I'll change property's phpDoc, though.

Reply:
I actually wonedr about whole idea here.
I understand the need to disable the test on MacOS.
but I wonder if anyone would ever use this `requirements['os']` other than this single performance test, that is failing due to CI problems and not due to MacOS itself.

Reply:
I've added array shape representing requirements, but since these are different I've changed return type of `getRequirement()` to `mixed` because different keys have have different values. I could make nested conditional return type, but it would be hard to maintain - as I said this method should not be so generic, or it should return some object representing requirements, with getters for each available requirement (with proper return type).